cmd/factory_test.go

```diff
@@ -0,0 +1,58 @@
+package cmd
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/mitchellh/go-homedir"
+)
+
+func TestEnvPathFactory(t *testing.T) {
+	path := "fake_path"
+	home, err := homedir.Dir()
+	if err != nil {
+		t.Errorf("Failed to find the home directory: %s", err.Error())
+	}
+
+	tests := []struct {
+		setQRI     bool
+		setIPFS    bool
+		path       string
+		qriAnswer  string
+		ipfsAnswer string
+	}{
+		{false, false, path, filepath.Join(home, ".qri"), filepath.Join(home, ".ipfs")},
+		{true, false, path, path, filepath.Join(home, ".ipfs")},
+		{false, true, path, path, path},
+	}
+
+	//Note: Variables created with os.Setenv exist for the rest of the process
+	for _, test := range tests {
+		if test.setQRI {
+			err := os.Setenv("QRI_PATH", test.path)
+			if err != nil {
+				t.Errorf("Failed to set up QRI_PATH: %s", err.Error())
+			}
+		}
+
+		if test.setIPFS {
+			err := os.Setenv("IPFS_PATH", test.path)
+			if err != nil {
+				t.Errorf("Failed to set up IPFS_PATH: %s", err.Error())
+			}
+		}
+
+		qriResult, ipfsResult := EnvPathFactory()
+
+		if !strings.EqualFold(qriResult, test.qriAnswer) {
+			t.Errorf("Expected qri path to be %s, but got %s", test.qriAnswer, qriResult)
```

Comment:
I'm a big fan of noting the case iteration number on these:
```golang
    t.Errorf("case %d Expected qri path to be %s, but got %s", i, test.qriAnswer, qriResult)
```
it makes our tests a little easier to debug 🙂

Reply:
Okay. Will do